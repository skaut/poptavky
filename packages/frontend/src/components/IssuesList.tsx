/** @jsxImportSource @emotion/react */
import { css } from "@emotion/react";
import type React from "react";
import ReactMarkdown from "react-markdown";
import { Link } from "react-router-dom";
import remarkGfm from "remark-gfm";

import type { Project } from "../interfaces/Project";
import type { ProjectInfo } from "../interfaces/ProjectInfo";
import type { ProjectIssue } from "../interfaces/ProjectIssue";
import type { ProjectListing } from "../interfaces/ProjectListing";
import { theme } from "../theme";
import { getIssueLink } from "../utils/getIssueLink";
import { H3 } from "./Typography";

export const IssuesList = ({
  issues,
  project,
}: {
  readonly issues: Array<ProjectIssue>;
  readonly project: ProjectListing | (Project & ProjectInfo);
}): React.JSX.Element => (
  <>
    {issues.map((issue) => (
      <article
        css={css`
          margin: 12px 0 24px;
          img {
            max-width: 100%;
          }
        `}
        key={issue.number}
      >
        <H3>
          <Link
            to={getIssueLink({
              project: {
                ...("info" in project
                  ? { ...project, ...project.info }
                  : { ...project }),
              },
              ...issue,
            })}
          >
            {issue.title}
          </Link>
          &nbsp;
          <span
            css={css`
              color: ${theme.colors.gray};
              font-weight: normal;
            `}
          >
            #{issue.number}
          </span>
        </H3>
        <ReactMarkdown remarkPlugins={[remarkGfm]}>
          {issue.description}
        </ReactMarkdown>
      </article>
    ))}
  </>
);
