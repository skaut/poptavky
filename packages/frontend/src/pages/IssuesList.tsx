/** @jsxImportSource @emotion/react */
import type React from "react"

import { Issue } from "../components/Issue"
import type { ProjectListings } from "../interfaces/ProjectListings"
import { getIssuesWithProjectInfo } from "../utils/getAllIssues"

export const IssuesList = ({
  data,
}: {
  readonly data: ProjectListings
}): React.JSX.Element => {
  const issues = getIssuesWithProjectInfo(data)
  return (
    <>
      {issues.map((issue) => (
        <Issue issue={issue} key={issue.link} />
      ))}
    </>
  )
}
