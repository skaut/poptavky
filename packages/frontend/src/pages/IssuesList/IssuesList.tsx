/** @jsxImportSource @emotion/react */
import { getIssuesWithProjectInfo } from "../../utils/getAllIssues"
import { Issue } from "../../components/Issue"
import { ProjectListings } from "../../interfaces/ProjectListings"

export const IssuesList = ({ data }: { data: ProjectListings }) => {
  const issues = getIssuesWithProjectInfo(data)
  return (
    <>
      {issues.map((issue) => (
        <Issue key={issue.link} issue={issue} />
      ))}
    </>
  )
}
