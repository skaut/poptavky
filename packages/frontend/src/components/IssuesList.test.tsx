import { MemoryRouter } from "react-router-dom"
import renderer from "react-test-renderer"

import { testData } from "../testData"
import { IssuesList } from "./IssuesList"

const project = testData.projects[0]

describe("IssuesList component", () => {
  test("should render correctly", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <IssuesList issues={project.issues} project={project} />
        </MemoryRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  test("should render correctly without link", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <IssuesList
            issues={project.issues.map((issue) => ({
              ...issue,
              link: undefined,
            }))}
            project={project}
          />
        </MemoryRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
