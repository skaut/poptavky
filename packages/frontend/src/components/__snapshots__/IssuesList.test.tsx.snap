// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssuesList component should render correctly 1`] = `
Array [
  .emotion-0 {
  margin: 12px 0 24px;
}

.emotion-0 img {
  max-width: 100%;
}

.emotion-1 {
  color: #294885;
  margin: 0 0 12px;
  font-family: themix;
}

.emotion-2 {
  color: #999;
  font-weight: normal;
}

<article
    className="emotion-0"
  >
    <h3
      className="emotion-1"
    >
      <a
        href="/skaut/skaut-google-drive-gallery/1"
        onClick={[Function]}
      >
        Add hint to set videos to public
      </a>
       
      <span
        className="emotion-2"
      >
        #
        1
      </span>
    </h3>
    <p>
      When editing and selecting a folder with videos, the editor could display a notice that setting videos to "Anyone can view" will speed them up.
    </p>
    

    <p>
      Depends on #792
    </p>
  </article>,
  .emotion-0 {
  margin: 12px 0 24px;
}

.emotion-0 img {
  max-width: 100%;
}

.emotion-1 {
  color: #294885;
  margin: 0 0 12px;
  font-family: themix;
}

.emotion-2 {
  color: #999;
  font-weight: normal;
}

<article
    className="emotion-0"
  >
    <h3
      className="emotion-1"
    >
      <a
        href="/skaut/skaut-google-drive-gallery/2"
        onClick={[Function]}
      >
        If a shortcut name differs from the original, the name in gallery and breadcrumbs differs
      </a>
       
      <span
        className="emotion-2"
      >
        #
        2
      </span>
    </h3>
    <p>
      When in a situation that there is a Google Drive shortcut somewhere in the gallery, the plugin displays the folder and it works for browsing. However, when looking at the shortcut as a member of its parent directory and when being inside the directory and looking at its name in the breadcrumbs, one uses the shortcut name, whereas the other uses the original folder name (when they differ).
    </p>
    

    <p>
      This is a bug and both should show the shortcut name.
    </p>
  </article>,
  .emotion-0 {
  margin: 12px 0 24px;
}

.emotion-0 img {
  max-width: 100%;
}

.emotion-1 {
  color: #294885;
  margin: 0 0 12px;
  font-family: themix;
}

.emotion-2 {
  color: #999;
  font-weight: normal;
}

<article
    className="emotion-0"
  >
    <h3
      className="emotion-1"
    >
      <a
        href="/skaut/skaut-google-drive-gallery/3"
        onClick={[Function]}
      >
        Editing doesn't check path validity
      </a>
       
      <span
        className="emotion-2"
      >
        #
        3
      </span>
    </h3>
    <p>
      When opening the editor which contains a gallery that is made from a folder that has been renamed or deleted, the editor breaks. It shows: 
      <code>
        Directory &quot;Galerie ODYWEB&quot; wasn&#039;t found - it may have been deleted or renamed.
      </code>
       Instead, it should verify that the path is valid and if it's not, should display a warning (and possibly move you to either the root directory, or the closest valid ancestor - however, this should not lead to you accidentally changing the gallery!)
    </p>
    

    <p>
      Last tested on 36dc1a10a353ba5bf2b98f33b49b1135a36ae3d3
    </p>
  </article>,
  .emotion-0 {
  margin: 12px 0 24px;
}

.emotion-0 img {
  max-width: 100%;
}

.emotion-1 {
  color: #294885;
  margin: 0 0 12px;
  font-family: themix;
}

.emotion-2 {
  color: #999;
  font-weight: normal;
}

<article
    className="emotion-0"
  >
    <h3
      className="emotion-1"
    >
      <a
        href="/skaut/skaut-google-drive-gallery/4"
        onClick={[Function]}
      >
        Ctrl-click on folders does not work
      </a>
       
      <span
        className="emotion-2"
      >
        #
        4
      </span>
    </h3>
    <p>
      Ctrl+click on a gallery item (subfolder) or breadcrumbs should open in a new tab
    </p>
  </article>,
  .emotion-0 {
  margin: 12px 0 24px;
}

.emotion-0 img {
  max-width: 100%;
}

.emotion-1 {
  color: #294885;
  margin: 0 0 12px;
  font-family: themix;
}

.emotion-2 {
  color: #999;
  font-weight: normal;
}

<article
    className="emotion-0"
  >
    <h3
      className="emotion-1"
    >
      <a
        href="/skaut/skaut-google-drive-gallery/5"
        onClick={[Function]}
      >
        Add option to specify gallery by a Drive link
      </a>
       
      <span
        className="emotion-2"
      >
        #
        5
      </span>
    </h3>
    <p>
      <a
        href="https://wordpress.org/support/topic/folder-link-instead-of-folder-name/"
      >
        https://wordpress.org/support/topic/folder-link-instead-of-folder-name/
      </a>
    </p>
    

    <p>
      Basically, instead of searching for the folder in WP, you'd just paste the link to it into the editor and the plugin would figure out the path (Google Drive files now return a list of parents, so it could be done.)
    </p>
    

    <p>
      Depends on #111.
    </p>
  </article>,
  .emotion-0 {
  margin: 12px 0 24px;
}

.emotion-0 img {
  max-width: 100%;
}

.emotion-1 {
  color: #294885;
  margin: 0 0 12px;
  font-family: themix;
}

.emotion-2 {
  color: #999;
  font-weight: normal;
}

<article
    className="emotion-0"
  >
    <h3
      className="emotion-1"
    >
      <a
        href="/skaut/skaut-google-drive-gallery/6"
        onClick={[Function]}
      >
        Add images to the block
      </a>
       
      <span
        className="emotion-2"
      >
        #
        6
      </span>
    </h3>
    <p>
      Currently, images aren't listed in the editor - both block and TinyMCE plugin. This often confuses users.
    </p>
  </article>,
  .emotion-0 {
  margin: 12px 0 24px;
}

.emotion-0 img {
  max-width: 100%;
}

.emotion-1 {
  color: #294885;
  margin: 0 0 12px;
  font-family: themix;
}

.emotion-2 {
  color: #999;
  font-weight: normal;
}

<article
    className="emotion-0"
  >
    <h3
      className="emotion-1"
    >
      <a
        href="/skaut/skaut-google-drive-gallery/7"
        onClick={[Function]}
      >
        Add video overlay
      </a>
       
      <span
        className="emotion-2"
      >
        #
        7
      </span>
    </h3>
    <p>
      In the image grid, we should add an overlay to signify that an item is a video (for example a big play icon).
    </p>
  </article>,
  .emotion-0 {
  margin: 12px 0 24px;
}

.emotion-0 img {
  max-width: 100%;
}

.emotion-1 {
  color: #294885;
  margin: 0 0 12px;
  font-family: themix;
}

.emotion-2 {
  color: #999;
  font-weight: normal;
}

<article
    className="emotion-0"
  >
    <h3
      className="emotion-1"
    >
      <a
        href="/skaut/skaut-google-drive-gallery/8"
        onClick={[Function]}
      >
        Increase test coverage
      </a>
       
      <span
        className="emotion-2"
      >
        #
        8
      </span>
    </h3>
    <p>
      <a
        href="https://codecov.io/gh/skaut/skaut-google-drive-gallery"
      >
        https://codecov.io/gh/skaut/skaut-google-drive-gallery
      </a>
    </p>
  </article>,
  .emotion-0 {
  margin: 12px 0 24px;
}

.emotion-0 img {
  max-width: 100%;
}

.emotion-1 {
  color: #294885;
  margin: 0 0 12px;
  font-family: themix;
}

.emotion-2 {
  color: #999;
  font-weight: normal;
}

<article
    className="emotion-0"
  >
    <h3
      className="emotion-1"
    >
      <a
        href="/skaut/skaut-google-drive-gallery/9"
        onClick={[Function]}
      >
        Themes with "editor-styles" support change the look of the block editor
      </a>
       
      <span
        className="emotion-2"
      >
        #
        9
      </span>
    </h3>
    <p>
      If the active theme contains
    </p>
    

    <pre>
      <code
        className="language-php"
      >
        add_theme_support( 'editor-styles' );

      </code>
    </pre>
    

    <p>
      its styles are also applied to the block editor, which then looks weird. We should implement our own styles for the editor.
    </p>
    

    <p>
      Example of such theme is the Twenty Seventeen theme.
    </p>
  </article>,
  .emotion-0 {
  margin: 12px 0 24px;
}

.emotion-0 img {
  max-width: 100%;
}

.emotion-1 {
  color: #294885;
  margin: 0 0 12px;
  font-family: themix;
}

.emotion-2 {
  color: #999;
  font-weight: normal;
}

<article
    className="emotion-0"
  >
    <h3
      className="emotion-1"
    >
      <a
        href="/skaut/skaut-google-drive-gallery/10"
        onClick={[Function]}
      >
        Google app creation manual is outdated
      </a>
       
      <span
        className="emotion-2"
      >
        #
        10
      </span>
    </h3>
    <p>
      <a
        href="https://napoveda.skaut.cz/dobryweb/en-skaut-google-drive-gallery/en-get-google-application"
      >
        https://napoveda.skaut.cz/dobryweb/en-skaut-google-drive-gallery/en-get-google-application
      </a>
      

      <a
        href="https://napoveda.skaut.cz/dobryweb/cs-skaut-google-drive-gallery/cs-ziskani-google-aplikace"
      >
        https://napoveda.skaut.cz/dobryweb/cs-skaut-google-drive-gallery/cs-ziskani-google-aplikace
      </a>
    </p>
    

    <p>
      Google changed the consent screen layout and require some form of verification.
    </p>
  </article>,
  .emotion-0 {
  margin: 12px 0 24px;
}

.emotion-0 img {
  max-width: 100%;
}

.emotion-1 {
  color: #294885;
  margin: 0 0 12px;
  font-family: themix;
}

.emotion-2 {
  color: #999;
  font-weight: normal;
}

<article
    className="emotion-0"
  >
    <h3
      className="emotion-1"
    >
      <a
        href="/skaut/skaut-google-drive-gallery/11"
        onClick={[Function]}
      >
        Directory preview images
      </a>
       
      <span
        className="emotion-2"
      >
        #
        11
      </span>
    </h3>
    <p>
      Currently, the first image in a directory is used for its preview.
    </p>
  </article>,
  .emotion-0 {
  margin: 12px 0 24px;
}

.emotion-0 img {
  max-width: 100%;
}

.emotion-1 {
  color: #294885;
  margin: 0 0 12px;
  font-family: themix;
}

.emotion-2 {
  color: #999;
  font-weight: normal;
}

<article
    className="emotion-0"
  >
    <h3
      className="emotion-1"
    >
      <a
        href="/skaut/skaut-google-drive-gallery/12"
        onClick={[Function]}
      >
        UX & design audit
      </a>
       
      <span
        className="emotion-2"
      >
        #
        12
      </span>
    </h3>
    <p>
      Go over the design of:
    </p>
    

    <ul
      className="contains-task-list"
    >
      

      <li
        className="task-list-item"
      >
        <input
          checked={false}
          disabled={true}
          type="checkbox"
        />
         
        gallery view 
        <a
          href="https://github.com/skaut/skaut-google-drive-gallery/issues/36"
        >
          https://github.com/skaut/skaut-google-drive-gallery/issues/36
        </a>
      </li>
      

      <li
        className="task-list-item"
      >
        <input
          checked={false}
          disabled={true}
          type="checkbox"
        />
         
        preview
      </li>
      

      <li
        className="task-list-item"
      >
        <input
          checked={false}
          disabled={true}
          type="checkbox"
        />
         
        folder thumbnail - overlay, folder title and image count
      </li>
      

      <li
        className="task-list-item"
      >
        <input
          checked={false}
          disabled={true}
          type="checkbox"
        />
         
        svg folder icon
      </li>
      

      <li
        className="task-list-item"
      >
        <input
          checked={false}
          disabled={true}
          type="checkbox"
        />
         
        gallery breadcrumbs
      </li>
      

      <li
        className="task-list-item"
      >
        <input
          checked={false}
          disabled={true}
          type="checkbox"
        />
         
        options page
      </li>
      

      <li
        className="task-list-item"
      >
        <input
          checked={false}
          disabled={true}
          type="checkbox"
        />
         
        WP.org plugin page
      </li>
      

      <li
        className="task-list-item"
      >
        <input
          checked={false}
          disabled={true}
          type="checkbox"
        />
         
        TinyMCE & Gutenberg editor plugins
      </li>
      

      <li
        className="task-list-item"
      >
        <input
          checked={false}
          disabled={true}
          type="checkbox"
        />
         
        TinyMCE & Gutenberg "Authentication failed" dialogs
      </li>
      

    </ul>
    

    <p>
      Everything on the frontend needs to work at least with the default WP template and DSW templates.
    </p>
  </article>,
]
`;

exports[`IssuesList component should render correctly without link 1`] = `
Array [
  .emotion-0 {
  margin: 12px 0 24px;
}

.emotion-0 img {
  max-width: 100%;
}

.emotion-1 {
  color: #294885;
  margin: 0 0 12px;
  font-family: themix;
}

.emotion-2 {
  color: #999;
  font-weight: normal;
}

<article
    className="emotion-0"
  >
    <h3
      className="emotion-1"
    >
      <a
        href="/skaut/skaut-google-drive-gallery/1"
        onClick={[Function]}
      >
        Add hint to set videos to public
      </a>
       
      <span
        className="emotion-2"
      >
        #
        1
      </span>
    </h3>
    <p>
      When editing and selecting a folder with videos, the editor could display a notice that setting videos to "Anyone can view" will speed them up.
    </p>
    

    <p>
      Depends on #792
    </p>
  </article>,
  .emotion-0 {
  margin: 12px 0 24px;
}

.emotion-0 img {
  max-width: 100%;
}

.emotion-1 {
  color: #294885;
  margin: 0 0 12px;
  font-family: themix;
}

.emotion-2 {
  color: #999;
  font-weight: normal;
}

<article
    className="emotion-0"
  >
    <h3
      className="emotion-1"
    >
      <a
        href="/skaut/skaut-google-drive-gallery/2"
        onClick={[Function]}
      >
        If a shortcut name differs from the original, the name in gallery and breadcrumbs differs
      </a>
       
      <span
        className="emotion-2"
      >
        #
        2
      </span>
    </h3>
    <p>
      When in a situation that there is a Google Drive shortcut somewhere in the gallery, the plugin displays the folder and it works for browsing. However, when looking at the shortcut as a member of its parent directory and when being inside the directory and looking at its name in the breadcrumbs, one uses the shortcut name, whereas the other uses the original folder name (when they differ).
    </p>
    

    <p>
      This is a bug and both should show the shortcut name.
    </p>
  </article>,
  .emotion-0 {
  margin: 12px 0 24px;
}

.emotion-0 img {
  max-width: 100%;
}

.emotion-1 {
  color: #294885;
  margin: 0 0 12px;
  font-family: themix;
}

.emotion-2 {
  color: #999;
  font-weight: normal;
}

<article
    className="emotion-0"
  >
    <h3
      className="emotion-1"
    >
      <a
        href="/skaut/skaut-google-drive-gallery/3"
        onClick={[Function]}
      >
        Editing doesn't check path validity
      </a>
       
      <span
        className="emotion-2"
      >
        #
        3
      </span>
    </h3>
    <p>
      When opening the editor which contains a gallery that is made from a folder that has been renamed or deleted, the editor breaks. It shows: 
      <code>
        Directory &quot;Galerie ODYWEB&quot; wasn&#039;t found - it may have been deleted or renamed.
      </code>
       Instead, it should verify that the path is valid and if it's not, should display a warning (and possibly move you to either the root directory, or the closest valid ancestor - however, this should not lead to you accidentally changing the gallery!)
    </p>
    

    <p>
      Last tested on 36dc1a10a353ba5bf2b98f33b49b1135a36ae3d3
    </p>
  </article>,
  .emotion-0 {
  margin: 12px 0 24px;
}

.emotion-0 img {
  max-width: 100%;
}

.emotion-1 {
  color: #294885;
  margin: 0 0 12px;
  font-family: themix;
}

.emotion-2 {
  color: #999;
  font-weight: normal;
}

<article
    className="emotion-0"
  >
    <h3
      className="emotion-1"
    >
      <a
        href="/skaut/skaut-google-drive-gallery/4"
        onClick={[Function]}
      >
        Ctrl-click on folders does not work
      </a>
       
      <span
        className="emotion-2"
      >
        #
        4
      </span>
    </h3>
    <p>
      Ctrl+click on a gallery item (subfolder) or breadcrumbs should open in a new tab
    </p>
  </article>,
  .emotion-0 {
  margin: 12px 0 24px;
}

.emotion-0 img {
  max-width: 100%;
}

.emotion-1 {
  color: #294885;
  margin: 0 0 12px;
  font-family: themix;
}

.emotion-2 {
  color: #999;
  font-weight: normal;
}

<article
    className="emotion-0"
  >
    <h3
      className="emotion-1"
    >
      <a
        href="/skaut/skaut-google-drive-gallery/5"
        onClick={[Function]}
      >
        Add option to specify gallery by a Drive link
      </a>
       
      <span
        className="emotion-2"
      >
        #
        5
      </span>
    </h3>
    <p>
      <a
        href="https://wordpress.org/support/topic/folder-link-instead-of-folder-name/"
      >
        https://wordpress.org/support/topic/folder-link-instead-of-folder-name/
      </a>
    </p>
    

    <p>
      Basically, instead of searching for the folder in WP, you'd just paste the link to it into the editor and the plugin would figure out the path (Google Drive files now return a list of parents, so it could be done.)
    </p>
    

    <p>
      Depends on #111.
    </p>
  </article>,
  .emotion-0 {
  margin: 12px 0 24px;
}

.emotion-0 img {
  max-width: 100%;
}

.emotion-1 {
  color: #294885;
  margin: 0 0 12px;
  font-family: themix;
}

.emotion-2 {
  color: #999;
  font-weight: normal;
}

<article
    className="emotion-0"
  >
    <h3
      className="emotion-1"
    >
      <a
        href="/skaut/skaut-google-drive-gallery/6"
        onClick={[Function]}
      >
        Add images to the block
      </a>
       
      <span
        className="emotion-2"
      >
        #
        6
      </span>
    </h3>
    <p>
      Currently, images aren't listed in the editor - both block and TinyMCE plugin. This often confuses users.
    </p>
  </article>,
  .emotion-0 {
  margin: 12px 0 24px;
}

.emotion-0 img {
  max-width: 100%;
}

.emotion-1 {
  color: #294885;
  margin: 0 0 12px;
  font-family: themix;
}

.emotion-2 {
  color: #999;
  font-weight: normal;
}

<article
    className="emotion-0"
  >
    <h3
      className="emotion-1"
    >
      <a
        href="/skaut/skaut-google-drive-gallery/7"
        onClick={[Function]}
      >
        Add video overlay
      </a>
       
      <span
        className="emotion-2"
      >
        #
        7
      </span>
    </h3>
    <p>
      In the image grid, we should add an overlay to signify that an item is a video (for example a big play icon).
    </p>
  </article>,
  .emotion-0 {
  margin: 12px 0 24px;
}

.emotion-0 img {
  max-width: 100%;
}

.emotion-1 {
  color: #294885;
  margin: 0 0 12px;
  font-family: themix;
}

.emotion-2 {
  color: #999;
  font-weight: normal;
}

<article
    className="emotion-0"
  >
    <h3
      className="emotion-1"
    >
      <a
        href="/skaut/skaut-google-drive-gallery/8"
        onClick={[Function]}
      >
        Increase test coverage
      </a>
       
      <span
        className="emotion-2"
      >
        #
        8
      </span>
    </h3>
    <p>
      <a
        href="https://codecov.io/gh/skaut/skaut-google-drive-gallery"
      >
        https://codecov.io/gh/skaut/skaut-google-drive-gallery
      </a>
    </p>
  </article>,
  .emotion-0 {
  margin: 12px 0 24px;
}

.emotion-0 img {
  max-width: 100%;
}

.emotion-1 {
  color: #294885;
  margin: 0 0 12px;
  font-family: themix;
}

.emotion-2 {
  color: #999;
  font-weight: normal;
}

<article
    className="emotion-0"
  >
    <h3
      className="emotion-1"
    >
      <a
        href="/skaut/skaut-google-drive-gallery/9"
        onClick={[Function]}
      >
        Themes with "editor-styles" support change the look of the block editor
      </a>
       
      <span
        className="emotion-2"
      >
        #
        9
      </span>
    </h3>
    <p>
      If the active theme contains
    </p>
    

    <pre>
      <code
        className="language-php"
      >
        add_theme_support( 'editor-styles' );

      </code>
    </pre>
    

    <p>
      its styles are also applied to the block editor, which then looks weird. We should implement our own styles for the editor.
    </p>
    

    <p>
      Example of such theme is the Twenty Seventeen theme.
    </p>
  </article>,
  .emotion-0 {
  margin: 12px 0 24px;
}

.emotion-0 img {
  max-width: 100%;
}

.emotion-1 {
  color: #294885;
  margin: 0 0 12px;
  font-family: themix;
}

.emotion-2 {
  color: #999;
  font-weight: normal;
}

<article
    className="emotion-0"
  >
    <h3
      className="emotion-1"
    >
      <a
        href="/skaut/skaut-google-drive-gallery/10"
        onClick={[Function]}
      >
        Google app creation manual is outdated
      </a>
       
      <span
        className="emotion-2"
      >
        #
        10
      </span>
    </h3>
    <p>
      <a
        href="https://napoveda.skaut.cz/dobryweb/en-skaut-google-drive-gallery/en-get-google-application"
      >
        https://napoveda.skaut.cz/dobryweb/en-skaut-google-drive-gallery/en-get-google-application
      </a>
      

      <a
        href="https://napoveda.skaut.cz/dobryweb/cs-skaut-google-drive-gallery/cs-ziskani-google-aplikace"
      >
        https://napoveda.skaut.cz/dobryweb/cs-skaut-google-drive-gallery/cs-ziskani-google-aplikace
      </a>
    </p>
    

    <p>
      Google changed the consent screen layout and require some form of verification.
    </p>
  </article>,
  .emotion-0 {
  margin: 12px 0 24px;
}

.emotion-0 img {
  max-width: 100%;
}

.emotion-1 {
  color: #294885;
  margin: 0 0 12px;
  font-family: themix;
}

.emotion-2 {
  color: #999;
  font-weight: normal;
}

<article
    className="emotion-0"
  >
    <h3
      className="emotion-1"
    >
      <a
        href="/skaut/skaut-google-drive-gallery/11"
        onClick={[Function]}
      >
        Directory preview images
      </a>
       
      <span
        className="emotion-2"
      >
        #
        11
      </span>
    </h3>
    <p>
      Currently, the first image in a directory is used for its preview.
    </p>
  </article>,
  .emotion-0 {
  margin: 12px 0 24px;
}

.emotion-0 img {
  max-width: 100%;
}

.emotion-1 {
  color: #294885;
  margin: 0 0 12px;
  font-family: themix;
}

.emotion-2 {
  color: #999;
  font-weight: normal;
}

<article
    className="emotion-0"
  >
    <h3
      className="emotion-1"
    >
      <a
        href="/skaut/skaut-google-drive-gallery/12"
        onClick={[Function]}
      >
        UX & design audit
      </a>
       
      <span
        className="emotion-2"
      >
        #
        12
      </span>
    </h3>
    <p>
      Go over the design of:
    </p>
    

    <ul
      className="contains-task-list"
    >
      

      <li
        className="task-list-item"
      >
        <input
          checked={false}
          disabled={true}
          type="checkbox"
        />
         
        gallery view 
        <a
          href="https://github.com/skaut/skaut-google-drive-gallery/issues/36"
        >
          https://github.com/skaut/skaut-google-drive-gallery/issues/36
        </a>
      </li>
      

      <li
        className="task-list-item"
      >
        <input
          checked={false}
          disabled={true}
          type="checkbox"
        />
         
        preview
      </li>
      

      <li
        className="task-list-item"
      >
        <input
          checked={false}
          disabled={true}
          type="checkbox"
        />
         
        folder thumbnail - overlay, folder title and image count
      </li>
      

      <li
        className="task-list-item"
      >
        <input
          checked={false}
          disabled={true}
          type="checkbox"
        />
         
        svg folder icon
      </li>
      

      <li
        className="task-list-item"
      >
        <input
          checked={false}
          disabled={true}
          type="checkbox"
        />
         
        gallery breadcrumbs
      </li>
      

      <li
        className="task-list-item"
      >
        <input
          checked={false}
          disabled={true}
          type="checkbox"
        />
         
        options page
      </li>
      

      <li
        className="task-list-item"
      >
        <input
          checked={false}
          disabled={true}
          type="checkbox"
        />
         
        WP.org plugin page
      </li>
      

      <li
        className="task-list-item"
      >
        <input
          checked={false}
          disabled={true}
          type="checkbox"
        />
         
        TinyMCE & Gutenberg editor plugins
      </li>
      

      <li
        className="task-list-item"
      >
        <input
          checked={false}
          disabled={true}
          type="checkbox"
        />
         
        TinyMCE & Gutenberg "Authentication failed" dialogs
      </li>
      

    </ul>
    

    <p>
      Everything on the frontend needs to work at least with the default WP template and DSW templates.
    </p>
  </article>,
]
`;
